collections/_posts/2025/04/05/2025-04-01-improving-mesherys-ui-e2e-tests.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Improving Meshery's UI E2E Tests
+author: Meshery Maintainers
+date: 2025-04-04
+categories: 
+  - meshery
+  - open-source
+  - Playwright
+  - E2E testing
+featured-image: /assets/images/logos/meshery-logo-dark-text-side.svg
+---
+
+Hey Meshery Community 👋
+
+As we continue evolving Meshery’s end-to-end (E2E) UI test suite, we wanted to proactively communicate some updates and best practices around these tests.
+
+Our ongoing efforts are focused on **improving visibility, performance, and coverage**, while also **reducing noise** from flaky or unstable test cases. This work is essential to ensure our test suite continues to provide confidence without becoming a blocker to fast iteration.
+
+## Why You Might See Some Test Failures
+
+As we refine and improve the test suite, you may start noticing failures from time to time. Don’t panic—this might actually mean the test is working as intended! 🎉 Here’s what we suggest when this happens:
+
+---
+
```

Comment:
I wonder if there's an opportunity to link to this guide, https://docs.meshery.io/project/contributing/contributing-ui-tests, somewhere in the post.

Reply:
great callout. i added in resource section